platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/crypto/EnhancedKeyStoreLoaderTest.java

```diff
@@ -183,6 +185,7 @@ void keyStoreLoaderNegativeCase2Test(final String directoryName) throws IOExcept
         assertThat(keyDirectory).exists().isDirectory().isReadable().isNotEmptyDirectory();
 
         assertThat(loader).isNotNull();
+        assertThatCode(loader::migrate).doesNotThrowAnyException();
```

Comment:
You only have unit tests for the changes?

Reply:
@iwsimon, Each of the unit tests in EnhancedKeyStoreLoader are run in different scenarios based on the arguments to the unit test.  (1) With keys only in pfx files.  (2) with some pem files and some pfx files.  (3) with keys that are purely in pem files.  

Are there unit test scenarios beyond these that I am missing?

Reply:
In your test, you only validate doesNotThrowAnyException(). If the errorCount>0, you didn't throw any exceptions. Is this test enough to catch errorCount>0?

Reply:
Let me think about what I can do to create a scenario that triggers the roll back behavior.